src/qt/clientmodel.cpp

```diff
@@ -60,9 +60,11 @@ int ClientModel::getNumConnections(unsigned int flags) const
     return nNum;
 }
 
-int ClientModel::getNumBlocks() const
+int ClientModel::getNumBlocks(bool fLock) const
 {
-    LOCK(cs_main);
+    if (fLock)
+        LOCK(cs_main);
```

Comment:
Instead of a boolean argument (boolean arguments tend to make code unreadable), it is better done by having two functions:
```
int getNumBlocksWithLock() const
{
    LOCK(cs_main);
    return getNumBlocks();
}
int getNumBlocks() const
{
    return chainActive.Height();
}
```
But as we use recursive locks everywhere there should be no reason to. If you have the lock, taking it again i harmless.